hoodie-client/src/main/java/com/uber/hoodie/io/HoodieIOHandle.java

```diff
@@ -125,6 +128,25 @@ public void write(HoodieRecord record, Optional<IndexedRecord> insertValue) {
     // NO_OP
   }
 
+  /**
+   * Perform the actual writing of the given record into the backing file.
+   */
+  public void write(HoodieRecord record, Optional<IndexedRecord> avroRecord, Optional<Exception> exception) {
+    Optional recordMetadata = record.getData().getMetadata();
+    try {
+      if (exception.isPresent()) {
+        throw exception.get();
```

Comment:
Instead of throwing and catching exceptions which has some overhead, can you copy the catch block here

Reply:
The idea is that the number of such exceptions should be really low, ideally, this is the same piece of code that used to run before, throw and catch exceptions in error scenarios so would like to keep it the same.

Reply:
+1 I think we should just do a `exception.get() instanceof Throwable` instead

Reply:
Alright, done.